src/webots/vrml/WbNode.cpp

```diff
@@ -1080,28 +1161,39 @@ bool WbNode::exportNodeHeader(WbVrmlWriter &writer) const {
 }
 
 void WbNode::exportNodeFields(WbVrmlWriter &writer) const {
-  foreach (WbField *field, fields())
-    if (!field->isDeprecated() && ((field->isVrml() || writer.isProto()) && field->singleType() != WB_SF_NODE))
-      field->write(writer);
+  if (!writer.isUrdf()) {
+    foreach (WbField *field, fields()) {
+      if (!field->isDeprecated() && ((field->isVrml() || writer.isProto()) && field->singleType() != WB_SF_NODE))
+        field->write(writer);
+    }
+  }
```

Comment:
```suggestion
  if (writer.isUrdf())
    return;

  foreach (WbField *field, fields()) {
    if (!field->isDeprecated() && ((field->isVrml() || writer.isProto()) && field->singleType() != WB_SF_NODE))
      field->write(writer);
  }
```

We tend to use return as soon as possible instead of encapsulating everything in a condition.

Reply:
I am sorry for this and similar code style mistakes, we were often forced to use [MISRA C](https://en.wikipedia.org/wiki/MISRA_C) (e.g. function has to have only one return statement) and the old habits stick :)

Reply:
Don't worry, I know it is not easy to switch from one coding style to another one.